File: apps/demo-fe-ts-tanstackstart/app/lib/api/reports.ts
import { apiFetch } from "./client";
import type { PLReport } from "./types";

export function getPLReport(startDate: string, endDate: string, currency: string): Promise<PLReport> {
  const params = new URLSearchParams({
    startDate,
    endDate,
    currency,
  });
  return apiFetch<PLReport>(`/api/v1/reports/pl?${params}`);
}

File: apps/demo-fe-elixir-phoenix/lib/demo_fe_exph/api/reports.ex
defmodule DemoFeExph.Api.Reports do
  @moduledoc "Reporting API calls: profit and loss report."

  alias DemoFeExph.Api.Client

  @doc "Fetches the P&L report for the given date range and currency."
  def get_pl_report(token, start_date, end_date, currency \\ "USD") do
    query =
      URI.encode_query(%{
        start_date: start_date,
        end_date: end_date,
        currency: currency
      })

    Client.get("/api/v1/reports/pl?#{query}", token)
  end
end
